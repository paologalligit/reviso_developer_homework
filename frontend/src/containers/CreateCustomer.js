import React, { Component } from 'react';
import { Button } from "semantic-ui-react";

import AddCustomerModal from '../components/AddCustomerModal';

class CreateCustomer extends Component {
  constructor(props) {
    super(props);

    this.state = {
      openAddCustomerModal: false,
    };
  }

  toggleAddCustomerModal = e => {
    if (e) {
      e.preventDefault();
    }

    this.setState(state => ({ openAddCustomerModal: !state.openAddCustomerModal }));
  };

  render() {
    const { openAddCustomerModal } = this.state;
    const { userId } = this.props;

    return [
      <Button
        key="create-customer-button"
        onClick={this.toggleAddCustomerModal}
      >
        Create Customer
      </Button>,
      <AddCustomerModal
        key="create-customer-modal"
        onClose={this.toggleAddCustomerModal}
        open={openAddCustomerModal}
        userId={userId}
      />
    ];
  }
}

export default CreateCustomer;
