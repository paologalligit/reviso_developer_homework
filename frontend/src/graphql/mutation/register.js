import gql from 'graphql-tag';

export default gql`
  mutation(
    $name: String!
    $surname: String!
    $username: String!
    $email: String!
    $password: String!
    $birthdate: String!
    $country: String!
    $city: String!
    $address: String!
    $postal: Int!
  ) {
    registerUser(
      name: $name
      surname: $surname
      username: $username
      email: $email
      password: $password
      birthdate: $birthdate
      country: $country
      city: $city
      address: $address
      postal: $postal
    ) {
      ok
      errors {
        path
        message
      }
    }
  }
`;
