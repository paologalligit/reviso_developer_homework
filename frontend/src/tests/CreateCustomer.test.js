import 'jsdom-global/register';
import React from 'react';
import toJson from 'enzyme-to-json';
import { mount, render } from 'enzyme';
import { MockedProvider } from 'react-apollo/test-utils';

import CreateCustomer from '../containers/CreateCustomer';

describe('renders create customer', () => {
  it('matches snapshot', () => {
    const wrapper = render(
      <MockedProvider>
        <CreateCustomer />
      </MockedProvider>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe('open and close create customer modal', () => {
  it('opens anc closes the modal correctly', () => {
    const wrapper = mount(
      <MockedProvider mocks={[]}>
        <CreateCustomer />
      </MockedProvider>,
    );
    expect(wrapper).toHaveLength(1);
    const createCustomer = wrapper.find('CreateCustomer');
    const button = wrapper.find('Button').first();

    expect(createCustomer.instance().state.openAddCustomerModal).toBeFalsy();
    button.simulate('click');
    expect(createCustomer.instance().state.openAddCustomerModal).toBeTruthy();
    button.simulate('click');
    expect(createCustomer.instance().state.openAddCustomerModal).toBeFalsy();
  });
});
