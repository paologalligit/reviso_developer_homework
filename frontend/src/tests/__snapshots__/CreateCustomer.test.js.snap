// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders create customer matches snapshot 1`] = `
<button
  class="ui button"
>
  Create Customer
</button>
`;
